src/lib/vault/index.ts

```diff
@@ -0,0 +1,368 @@
+      path: `/${this.cfg.mounts.pki}/root/generate/exported`,
+      method: 'POST',
+      json: {
+        common_name: csr.CN,
+        ou: csr.OU,
+        organization: csr.O,
+        locality: csr.L,
+        country: csr.C,
+        province: csr.ST,
+        key_type: this.cfg.keyAlgorithm,
+        key_bits: this.cfg.keyLength,
+      },
+    });
+
+    return {
+      cert: data.certificate,
+      key: data.private_key,
+    };
+  }
+
+  async getCA() {
+    assert(this.client);
+    return this.client.request({
+      path: `/${this.cfg.mounts.pki}/ca/pem`,
+      method: 'GET',
+    });
+  }
+
+  async createDFSPServerCert(csrParameters: Record<string, any>) {
+    const reqJson: Record<string, any> = {
+      common_name: csrParameters.subject.CN,
+    };
+    if (csrParameters?.extensions?.subjectAltName) {
+      const { dns, ips } = csrParameters.extensions.subjectAltName;
+      if (dns) {
+        reqJson.alt_names = dns.join(',');
+      }
+      if (ips) {
+        reqJson.ip_sans = ips.join(',');
+      }
+    }
+    assert(this.client);
+    const { data } = await this.client.request({
+      path: `/${this.cfg.mounts.pki}/issue/${this.cfg.pkiServerRole}`,
+      method: 'POST',
+      json: reqJson,
+    });
+    return data;
+  }
+
+  /**
+   * Sign Hub CSR
+   * @param params
+   * @returns {Promise<*>}
+   */
+  async signHubCSR(csr: Record<string, any>) {
+    assert(this.client);
+    const { data } = await this.client.request({
+      path: `/${this.cfg.mounts.pki}/sign/${this.cfg.pkiClientRole}`,
+      method: 'POST',
+      json: {
+        common_name: this.cfg.commonName,
+        csr: csr,
+        // ttl: `${this._signExpiryHours}h`,
+      },
+    });
+    return data;
+  }
+
+  async setDFSPCaCertChain(certChainPem: string, privateKeyPem: string) {
+    assert(this.client);
+    await this.client.request({
+      path: `/${this.cfg.mounts.pki}/config/ca`,
+      method: 'POST',
+      json: {
+        pem_bundle: `${privateKeyPem}\n${certChainPem}`,
+      },
+    });
+    // Secret object documentation:
+    // https://github.com/modusintegration/mojaloop-k3s-bootstrap/blob/e3578fc57a024a41023c61cd365f382027b922bd/docs/README-vault.md#vault-crd-secrets-integration
+    // https://vault.koudingspawn.de/supported-secret-types/secret-type-cert
+  }
+
+  async getDFSPCaCertChain() {
+    assert(this.client);
+    return this.client.request({
+      path: `/${this.cfg.mounts.pki}/ca_chain`,
+      method: 'GET',
+    });
+  }
+
+  async setHubEndpoints(value: any) {
+    return this._setSecret(vaultPaths.HUB_ENDPOINTS, value);
+  }
+
+  async getHubEndpoints() {
+    return this._getSecret(vaultPaths.HUB_ENDPOINTS);
+  }
+
+  createCSR(keyBits: number, csrParameters: any = {}) {
+    const keys = forge.pki.rsa.generateKeyPair(keyBits);
+    const csr = forge.pki.createCertificationRequest();
+    csr.publicKey = keys.publicKey;
+    // if (csrParameters?.subject) {
```

Comment:
Do we need this?

Reply:
If you added this, please add a comment as to why its here.

If it was already, then that's fine we can resolve this for now.